tests/unit/test_sns_listener.py

```diff
@@ -120,6 +120,32 @@ def test_create_sns_message_body_json_structure_sqs_protocol(self):
 
         assert (result['Message'] == 'sqs message')
 
+    def test_create_sqs_message_attributes(self):
+        self.subscriber['RawMessageDelivery'] = 'true'
+        action = {
+            'Message': ['msg'],
+            'Subject': ['subject'],
+            'MessageAttributes.entry.1.Name': ['attr1'],
+            'MessageAttributes.entry.1.Value.DataType': ['String'],
+            'MessageAttributes.entry.1.Value.StringValue': ['value1'],
+            'MessageAttributes.entry.2.Name': ['attr2'],
+            'MessageAttributes.entry.2.Value.DataType': ['Binary'],
+            'MessageAttributes.entry.2.Value.BinaryValue': ['value2'.encode('utf-8')],
+            'MessageAttributes.entry.3.Name': ['attr3'],
+            'MessageAttributes.entry.3.Value.DataType': ['Number'],
+            'MessageAttributes.entry.3.Value.StringValue': ['value3'],
+        }
+
+        attributes = sns_listener.get_message_attributes(action)
+        result = sns_listener.create_sqs_message_attributes(self.subscriber, attributes)
+
+        assert (result['attr1']['DataType'] == 'String')
```

Comment:
nitpick: Better use `self.assertEqual` instead of `assert` (this also prints the actual value in case the assertion fails):
```
self.assertEqual(result['attr1']['DataType'], 'String')
...
```

If we could also refactor the existing assertions in this test class as part of this PR, that would be great. Thanks!